### ISSUE TITLE
UiSelect still doesn't work with preselect

### ISSUE THREAD

**Issue by EmilMoe** [JosephusPaye/Keen-UI#129]
I think you still have some issues with the `<ui-select>` default or value. I have associated a object with the same value, but it doesn't preselect it, nothing is selected, just like 0.8.8

**Comment by EmilMoe** [comment]
And if I give it a string as default I'm getting 

~~~~
Uncaught TypeError: Cannot read property 'length' of null

**Comment by EmilMoe** [comment]
After I updated to 0.8.9 I'm also receiving this error

~~~~
[Vue warn]: Error when evaluating expression "function filteredOptions() {
	            if (this.optionsDynamic) {
	                return this.options;
	            }
	
	            return this.options.filter(this.search);
	        }": TypeError: Cannot read property 'filter' of null (found in component: <ui-select>)

**Comment by EmilMoe** [comment]
Seems like the problem is that it doesn't wanna show any selection at all?

**Comment by JosephusPaye** [comment]
Hi @EmilMoe, I just used UiSelect recently and pre select worked for me. Could you create a quick demo showing the issue? 

Here's a Codepen template with the latest Keen UI and Vue: http://codepen.io/pen?template=NrKmPQ

**Comment by EmilMoe** [comment]
Thanks, @JosephusPaye 

I have created one here http://codepen.io/emilmoe/pen/bZJygE

**Comment by JosephusPaye** [comment]
I think I found what the problem is, first UiSelect expects the `text` key to be a string as we check the length here: https://github.com/JosephusPaye/Keen-UI/blob/gh-pages/src/UiSelect.vue#L181 (this is not intentional and will be fixed in v1.0, but that's not coming out soon).

Also, the default value should be set using the `default` prop and not with the `value` prop, since the `value` prop is overriden when the select is initialized.

See here for a working version of your demo: http://codepen.io/JosephusPaye/pen/pbBXbK?editors=1000

**Comment by EmilMoe** [comment]
I have created a pull request, would it be possible to apply this fix earlier? It shouldn't change any behaviour except that it makes sure that the tet is always a string.

**Comment by JosephusPaye** [comment]
Yeah, send in the pull and I'll see what I can do.

**Comment by victor-ponamariov** [comment]
Are we talking about default value for ui-select here? If so, then it doesn't work for me. All I did:
```
<ui-select name="customer"
               label="Customer"
               :value.sync="selectedCustomer"
               :show-search="true"
               :options="convertedCustomers"
               :default="convertedCustomers[0]"
               placeholder="Select Customer"
               validation-rules="required"
               @selected="customerSelected()"></ui-select>
```

Where `convertedCustomers` is an array of objects containing text/value. So the select is filled with this array but there is no pre-selected value

**Comment by JosephusPaye** [comment]
This should be fixed in v1.0. You can simply set the `value` prop and that will be used as the default value. 

Remember to use an array for `value` if `multiple` is `true`.
